torch/csrc/api/src/nn/module.cpp

```diff
@@ -62,6 +62,14 @@ const std::string& Module::name() const noexcept {
   // destroyed even if it is not the most-derived class.
   if (!name_.has_value()) {
     name_ = c10::demangle(typeid(*this).name());
+#if defined(_WIN32)
+    // Windows adds "struct" or "class" as a prefix.
```

Comment:
Ouch. any way to get around this special case? cc @davidbrownellMS and @peterjc123 . If not, so be it.

Reply:
[That](https://www.boost.org/doc/libs/1_66_0/boost/dll/detail/demangling/msvc.hpp)'s how Boost deals with this. So I think it should be fine.